src/main/host/process.rs

```diff
@@ -332,31 +461,32 @@ impl Process {
             id: process_id,
             host_id: host.id(),
             working_dir,
-            shim_shared_mem_block,
-            memory_manager: Box::new(RefCell::new(memory_manager)),
-            desc_table,
-            itimer_real,
             name,
             file_basename,
             plugin_name,
-            strace_logging,
-            dumpable: Cell::new(cshadow::SUID_DUMP_USER),
-            return_code: Cell::new(None),
-            #[cfg(feature = "perf_timers")]
-            cpu_delay_timer,
-            #[cfg(feature = "perf_timers")]
-            total_run_time: Cell::new(Duration::ZERO),
-            native_pid,
-            unsafe_borrow_mut: RefCell::new(None),
-            unsafe_borrows: RefCell::new(Vec::new()),
-            threads,
         };
         RootedRc::new(
             host.root(),
             RootedRefCell::new(
                 host.root(),
                 Self {
-                    state: RefCell::new(Some(ProcessState::Runnable(RunnableProcess { common }))),
+                    state: RefCell::new(Some(ProcessState::Runnable(RunnableProcess {
+                        common,
+                        shim_shared_mem_block,
+                        memory_manager: Box::new(RefCell::new(memory_manager)),
+                        desc_table,
+                        itimer_real,
+                        strace_logging,
+                        dumpable: Cell::new(cshadow::SUID_DUMP_USER),
+                        native_pid,
+                        unsafe_borrow_mut: RefCell::new(None),
+                        unsafe_borrows: RefCell::new(Vec::new()),
```

Comment:
Should these be left in `Common` so that we don't accidentally drop them while the C code still has borrows? Or are we sure that the process state will never become non-runnable while the C code still has borrows?

Reply:
> Should these be left in Common so that we don't accidentally drop them while the C code still has borrows? 

We'd then have to put the `memory_manager` in `Common` as well since these refer to it.

> Or are we sure that the process state will never become non-runnable while the C code still has borrows?

We only do the state change from in `Process`, where there shouldn't be any `Thread` or `SysCallHandler` code running, so there shouldn't be any outstanding borrows. I changed `RunnableProcess::into_common()` to assert that there are no remaining borrows instead of flushing them itself to validate this at runtime.

Ultimately it might make sense to pass a `&RunnableProcess` to thread methods and syscall handlers etc to make it clearer that it's guaranteed the process is running - that might let us also get rid of `NestedRef` etc. I started to sketch this out, but it's a big change to cram into this PR; e.g. we'd have to expose `RunnableProcess` to C code and change most of the C APIs to take it (or add ones that take it in case we still need Process in some places).